File: go_tanks/tank.go
package go_tanks

type Tank struct {
  Id            int
  Coords        *Coords
  LeftMotor     float64
  RightMotor    float64
}

func NewTank ( id int, coords *Coords ) *Tank {
  tank := Tank{
    Id: id,
    Coords: coords,
    LeftMotor: 0,
    RightMotor: 0,
  }
  return &tank
}

File: src/web/public/js/ng/tank.js
angular.module('app').service('$tank', function () {

  this.Id = 0
  this.Health = 0
  this.Coords = { X: 0, Y: 0 }
  this.Direction = 0
  this.LeftMotor = 0
  this.RightMotor = 0
  this.Gun = {
    Direction: 0
  }

});
